docs/tutorials/chatbot.md

```diff
@@ -0,0 +1,571 @@
+    )
+```
+
+Right now we haven't defined `MyTransformer` or `MyIndexer`. Let's create some dummy Executors so we can try our app. These will not be our final Executors but just something basic to learn first.
+
+### Create dummy Executors
+
+Now we have a Flow with two Executors. Write the following in your code:
+
+``` python
+from jina import Executor, requests
+class MyTransformer(Executor):
+    @requests(on='/foo')
+    def foo(self, **kwargs):
+        print(f'foo is doing cool stuff: {kwargs}')
+class MyIndexer(Executor):
+    @requests(on='/bar')
+    def bar(self, **kwargs):
+        print(f'bar is doing cool stuff: {kwargs}')
+```
+
+We will have more complex Executors later. For now our two Executors are only printing a line.
+
+So far it's a simple Flow but it is still useful to visualize it to make sure we have what we want.
+
+### Visualize a Flow
+
+By now, your code should look like this:
+
+``` python
+from jina import Flow, Document, Executor, requests
+class MyTransformer(Executor):
+    @requests(on='/foo')
+    def foo(self, **kwargs):
+        print(f'foo is doing cool stuff: {kwargs}')
+class MyIndexer(Executor):
+    @requests(on='/bar')
+    def bar(self, **kwargs):
+        print(f'bar is doing cool stuff: {kwargs}')
+flow = (
+        Flow()
+        .add(uses=MyTransformer)
+        .add(uses=MyIndexer)
+    )
+```
+
+If you want to visualize your Flow you can do that with `plot`. So add the `.plot()` function at the end of your Flow
+
+``` python
+from jina import Flow
+flow = (
+        Flow()
+        .add(uses=MyTransformer)
+        .add(uses=MyIndexer)
+        .plot('our_flow.svg')
+    )
+```
+
+Let's run the code we have so far. If you try it, not much will happen since we are not indexing anything yet, but you will see the new file `our_flow.svg` created on your working folder, and if you open it you would see this:
+
+```{figure} ../../.github/images/plot_flow1.svg
+:align: center
+```
+
+You can see a Flow with two Executors, but what if you have many Executors? this can quickly become very messy, so it is better to name the Executors with `name='CoolName'`. So in our example, we use:
+
+``` python
+from jina import Flow
+flow = (
+        Flow()
+        .add(name='MyTransformer', uses=MyTransformer)
+        .add(name='MyIndexer', uses=MyIndexer)
+        .plot('our_flow.svg')
+    )
+```
+
+Now if you run this, you should have a Flow that is more explicit:
+
+```{figure} ../../.github/images/plot_flow2.svg
+:align: center
+```
+
+
+### Use a Flow
+
+Ok, we have our Flow created and visualized. Let's put it to use now. The correct way to use a Flow is to open it as a context manager, using the `with` keyword:
+
+``` python
+with flow:
+    ...
+```
+
+Before we use it in our example, let's recap a bit of what we have seen:
+
+``` python
+from jina import Flow
+flow = Flow()          # Create Flow
+flow.add().add()       # Add elements to Flow
+flow.plot()            # Visualize a Flow
```

Comment:
Not needed anymore, no?

Reply:
I think the purpose of including these lines was to do a recapitulation. Do you think I should remove the whole recapitulation

Reply:
Ok, let's leave it here for now then. It just looks strange to me - kind of implies this is how your workflow should look like.

Reply:
I'll reformat it into a table, what do you think ?

Reply:
Good idea

Reply:
done